pkg/cloudprovider/aws/pricing.go

```diff
@@ -52,9 +53,13 @@ type PricingProvider struct {
 	onDemandUpdateTime time.Time
 	onDemandPrices     map[string]float64
 	spotUpdateTime     time.Time
-	spotPrices         map[string]float64
+	spotPrices         map[string]map[string]float64
 }
 
+// Default zone is used to store the pricing data based off of instance
+// type level information until zonal information is available
+const defaultZone = "default"
```

Comment:
I don't quite understand this, when is zonal information not available?

Reply:
I was thinking that if Karpenter's role didn't have permission on the SpotPriciningHistory API than we are never going to be able to get spot pricing information but we should still give back pricing based off of the on-demand pricing that we grabbed at startup. 

I would want to have static pricing data but we don't know what region this cluster is sitting in and therefore don't have specific data on the AZ-based data, unless we wanted to grab all the AZ-based data from all AZs across all regions. Seems like a lot of data when the instance will have permissions most of the time

Reply:
I guess alternatively the call to `Price()` itself can fallback to the initial on-demand data when the data isn't there for spot

Reply:
> I guess alternatively the call to Price() itself can fallback to the initial on-demand data when the data isn't there for spot

Exactly, and this is what's implemented on initialization right?